packages/core/src/getters/combine.ts

```diff
@@ -263,17 +270,40 @@ export const combineSchemas = ({
   };
 };
 
-const getCombineEnumValue = ({
-  values,
-  isRef,
-  originalSchema,
-}: CombinedData) => {
-  if (values.length === 1) {
-    if (isRef[0]) {
-      return values[0];
-    }
+const getEnumImplementation = (
+  enumValue: string,
+  enumName: string,
+  enumGenerationType: EnumGeneration,
+) => {
+  if (enumGenerationType === EnumGeneration.CONST)
+    return `// eslint-disable-next-line @typescript-eslint/no-redeclare\nexport const ${enumName} = {${enumValue}} as const`;
+  if (enumGenerationType === EnumGeneration.ENUM)
+    return `export enum ${enumName} {${enumValue}}`;
+  if (enumGenerationType === EnumGeneration.UNION)
+    return `export type ${enumName} = ${enumValue}`;
+  throw new Error(`Invalid enumGenerationType: ${enumGenerationType}`);
```

Comment:
This case will probably be excluded when normalizing.

```suggestion
```

Reply:
https://github.com/orval-labs/orval/pull/2057/commits/0b52d5662c3b0dbb51f9815b137451b6fa35d779